### PR TITLE
equalising devel and 3.4 in agency/cluster

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2515/

File: arangod/Cluster/SynchronizeShard.cpp [hunk 1]
```diff
@@ -941,6 +948,13 @@ ResultT<TRI_voc_tick_t> SynchronizeShard::catchupWithReadLock(
   double timeout = 300.0;
   TRI_voc_tick_t tickReached = 0;
   while (didTimeout && tries++ < 18) { // This will try to sync for at most 1 hour. (200 * 18 == 3600)
+    
+    if (isStopping()) {
+      std::string errorMessage = 
+        "synchronizeOneShard: startReadLockOnLeader (soft): shutting down";
+      return ResultT<TRI_voc_tick_t>::error(TRI_ERROR_INTERNAL, errorMessage);
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: should rather be `TRI_ERROR_SHUTTING_DOWN` than `TRI_ERROR_INTERNAL`

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2529/

**Comment by neunhoef** [comment]
LGTM

**Pull request merged**